test/e2e/e2e_test.go

```diff
@@ -133,6 +146,67 @@ var _ = Describe("controller", Ordered, func() {
 				return nil
 			}
 			EventuallyWithOffset(1, verifyOneRunningPod, time.Minute, time.Second).Should(Succeed())
+
+			By("validating that the nginx pod is running as expected")
+			//var nodeName string
+			verifyOneRunningNginxPod := func() error {
+				pods, err := clientset.CoreV1().Pods("ingress-nginx").List(ctx, v1.ListOptions{
+					LabelSelector: "app.kubernetes.io/component=controller", Limit: 1,
+				})
+				ExpectWithOffset(1, err).NotTo(HaveOccurred())
+				if len(pods.Items) != 1 {
+					return fmt.Errorf("got %d nginx pods", len(pods.Items))
+				}
+				if pods.Items[0].Status.Phase != "Running" {
+					return fmt.Errorf("nginx pod in %s status", pods.Items[0].Status.Phase)
+				}
+				fmt.Printf("nginx pod %s running on %s\n", pods.Items[0].Name, pods.Items[0].Spec.NodeName)
+				//nodeName = pods.Items[0].Spec.NodeName
+				return nil
+			}
+			EventuallyWithOffset(1, verifyOneRunningNginxPod, time.Minute, time.Second).Should(Succeed())
+			By("Verify PDB and PDBWatcher exist")
+			verifyPdbExists := func() error {
+				pdbList, err := clientset.PolicyV1().PodDisruptionBudgets("ingress-nginx").List(ctx,
+					v1.ListOptions{Limit: 1})
+				Expect(err).NotTo(HaveOccurred())
+				fmt.Printf("found %d pdbs in namespace ingress-nginx\n", len(pdbList.Items))
+				for _, pdb := range pdbList.Items {
+					fmt.Printf("found pdb name: %s \n", pdb.Name)
+					if pdb.Name != "ingress-nginx-controller" {
+						return fmt.Errorf("nginx pdb is not present on cluster")
+					}
+					if pdb.Spec.MinAvailable != nil {
+						if val := pdb.Spec.MinAvailable.IntValue(); val != 1 {
+							return fmt.Errorf("PDB '%s' has MinAvailable set to %d (not 1)\n", pdb.Name, val)
+						}
+						fmt.Printf("PDB '%s' has MinAvailable set to 1\n", pdb.Name)
+					}
+				}
+				return nil
+			}
+			verifyEvictionAutoScalerExists := func() error {
+				gvr := schema.GroupVersionResource{
+					Group:    "eviction-autoscaler.azure.com", // API group for your custom resource
+					Version:  "v1",                            // API version for the custom resource
+					Resource: "evictionautoscalers",           // Resource name (plural form of your custom resource)
+				}
+				evictionAutoScalerList, err := dynamicClient.Resource(gvr).Namespace("ingress-nginx").List(ctx,
```

Comment:
I don't think you actually need the dyanmic client. I think you can take the regular client and add eviction watcher to its scheme. Tyler has done this somewherfe in the aks-rp code base.